scripts/setupTypeScriptRollup.js

```diff
@@ -0,0 +1,135 @@
+// @ts-check
+
+/** This script modifies the project to support TS code in .svelte files like:
+  <script lang="ts">
+ 
+  As well as validating the code for CI.
+  */
+
+/**  To work on this script:
+  rm -rf test-template sapper-template && git clone sveltejs/sapper-template test-template && node scripts/setupTypeScript.js test-template
+*/
+
+const fs = require("fs")
+const path = require("path")
+const { argv } = require("process")
+
+const projectRoot = argv[2] || path.join(__dirname, "..")
+
+console.log("Adding typescript with rollup")
+
+if(fs.existsSync(path.join(projectRoot, "package.json"))) {
+
+  throw new Error("package.json already created. This script only works on plain new templates!")
+
+}
+
+// Add deps to pkg.json
+let pkgJSONPath = path.join(projectRoot, "package_template.json")
+const packageJSON = JSON.parse(fs.readFileSync(pkgJSONPath, "utf8"))
+packageJSON.devDependencies = Object.assign(packageJSON.devDependencies, {
+  "svelte-check": "^1.0.0",
+  "svelte-preprocess": "^4.0.0",
+  "@rollup/plugin-typescript": "^4.0.0",
+  "typescript": "^3.9.3",
+  "tslib": "^2.0.0",
+  "@tsconfig/svelte": "^1.0.0",
+  "@types/compression": "^1.7.0",
+  "@types/node": "^14.0.27",
+  "@types/polka": "^0.5.1"
+})
+
+// Add script for checking
+packageJSON.scripts = Object.assign(packageJSON.scripts, {
+  "validate": "svelte-check"
+})
+
+// Write the package JSON
+fs.writeFileSync(pkgJSONPath, JSON.stringify(packageJSON, null, "  "))
+
+scanFolderAndReplace(path.join(projectRoot, "src"))
+
+// replace js script tags to ts
+// replace js filenames to ts
+function scanFolderAndReplace (dir) {
+
+  const elements = fs.readdirSync(dir, { withFileTypes: true })
+
+  for(let i = 0; i < elements.length; i++) {
+
+    if(elements[i].isDirectory()) scanFolderAndReplace(path.join(dir, elements[i].name))
+    else if(elements[i].name.match(/.svelte/)) {
+
+      let svelteFile = fs.readFileSync(path.join(dir, elements[i].name), "utf8")
+      svelteFile = svelteFile.replace("<script", '<script lang="ts"')
```

Comment:
I think I'd remove this and use the default lang setting instead

Reply:
Default lang? You mean js?

Reply:
The svelte template also does add `lang="ts"` . I feel like both templates should be consistent on this.

Reply:
By default lang, I meant use this setting to specify that the templates are written in TypeScript: https://github.com/sveltejs/svelte-preprocess/blob/master/docs/getting-started.md#31-setting-default-languages

Good point about the templates remaining consistent. Would you be open to a PR to change the svelte template?

Reply:
I did know know this was a thing at all. I personally think it is better to be explicit about the language, if it is non-standard.

Reply:
I think I won't change it for now, unless they change it in the svelte template. Personally, I think `ts` is shorter, and I'm all in for simplicity